### PR TITLE
feat(api-service): plain cards fetch user organizations

### PR DESCRIPTION
**Pull request by jainpawan21**
### What changed? Why was the change needed?
- fetch user organizations for plain cards 

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/support/dto/plain-card.dto.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class PlainCutomer {
```
Comment (scopsy) [inline apps/api/src/app/support/dto/plain-card.dto.ts]:
```suggestion
export class PlainCustomer {
```

File: apps/api/src/app/support/dto/plain-card.dto.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export class PlainThread {
+  @ApiProperty()
+  id?: string;
+
+  @ApiProperty()
+  externalId?: string;
+}
+
+export class PlainCardRequestDto {
```
Comment (scopsy) [inline apps/api/src/app/support/dto/plain-card.dto.ts:30]: Are there any types available to export from their SDK?

File: apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { BaseCommand } from '@novu/application-generic';
+import { IsArray, IsDefined, IsOptional, IsString } from 'class-validator';
+
+export class PlainCutomer {
```
Comment (scopsy) [inline apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts]: There seem to be some duplication in the DTO's I would suggest simplifying this a bit, or reusing if they are identical, just for simplifying our maintenance.

File: apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts
Comment (scopsy) [inline apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts:1]: Should we consider renaming this usecases to: get-plain-cards.usecase.ts ? And then just have multiple functions for each card inside?

File: apps/api/src/app/support/support.controller.ts [hunk 4]
```diff
@@ -1,127 +1,22 @@
...
-                    },
-                  },
-                ],
-              },
-            },
-          ],
-        },
-      ],
-    };
+  async getUserOrganizations(@Body() body: PlainCardRequestDto) {
```
Comment (scopsy) [inline apps/api/src/app/support/support.controller.ts]: How authorization happening here? We need to verify the HMAC using their signing key and enforce it.

**Review by scopsy — CHANGES_REQUESTED** [review]
💪

File: apps/api/src/app/support/dto/plain-card.dto.ts (hunk 2 above)
Comment (jainpawan21) [reply apps/api/src/app/support/dto/plain-card.dto.ts:30]: No, their sdk do not export any type for card

File: apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts (hunk 3 above)
Comment (jainpawan21) [reply apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts]: reused dto in command

File: apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts
Comment (jainpawan21) [reply apps/api/src/app/support/usecases/fetch-user-organizations/fetch-user-organizations.command.ts:1]: renamed to plain-cards.usecase.ts

File: apps/api/src/app/support/support.controller.ts (hunk 4 above)
Comment (jainpawan21) [reply apps/api/src/app/support/support.controller.ts]: Added Plain hmac secret. linked PR https://github.com/novuhq/cloud-infra/pull/155

Comment (scopsy) [reply apps/api/src/app/support/support.controller.ts]: @jainpawan21 but where is the logic for the validation is happening? Maybe I'm missing something 🤔

Comment (scopsy) [reply apps/api/src/app/support/support.controller.ts]: NVM, found it 😅

File: apps/api/src/app/support/guards/plain-cards.guard.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import crypto from 'node:crypto';
+
+@Injectable()
+export class PlainCardsGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
+
+    const requestBody = JSON.stringify(request.body);
+    const plainCardsHMACSecretKey = process.env.PLAIN_CARDS_HMAC_SECRET_KEY as string;
+    const incomingSignature = request.headers['plain-request-signature'];
```
Comment (scopsy) [inline apps/api/src/app/support/guards/plain-cards.guard.ts:12]:
```suggestion
    const incomingSignature = request.headers['plain-request-signature'];
    if (!incomingSignature) throw new AuthError
```

File: apps/api/src/app/support/usecases/plain-cards.usecase.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+              {
+                componentText: {
+                  text: 'This user is not yet registered on Novu',
+                },
+              },
+            ],
+          },
+        ],
+      };
+    } else {
```
Comment (scopsy) [inline apps/api/src/app/support/usecases/plain-cards.usecase.ts]: nitpick: Else can be removed as we exist early in the previous part

**Review by scopsy — APPROVED** [review]

**Review by scopsy — APPROVED** [review]

**Pull request merged**